aarch64/kernel/nano_core/src/main.rs

```diff
@@ -0,0 +1,51 @@
+#![feature(naked_functions)]
+#![feature(abi_efiapi)]
+#![no_std]
+#![no_main]
+
+extern crate alloc;
+
+use alloc::vec;
+use core::fmt::Write;
+use core::arch::asm;
+
+use uefi_services::init;
+use uefi::prelude::entry;
+use uefi::Status;
+use uefi::Handle;
+use uefi::table::SystemTable;
+use uefi::table::Boot;
+
+use pl011_qemu::PL011;
+use pl011_qemu::UART1;
+
+pub type Pl011 = PL011<UART1>;
+
+#[inline(never)]
+extern "C" fn inf_loop_0xbeef() -> ! {
+    unsafe { asm!("mov x1, #0xbeef") };
+    loop {}
+}
+
+#[entry]
+fn main(
+    handle: Handle,
+    mut system_table: SystemTable<Boot>,
+) -> Status {
```

Comment:
Instead of unwrapping all the Results in this function, return them. It might be easiest to move everything below to a separate function that returns a Result such that you can use the try `?` operator, and then convert that `Result` into `Status` in this top-level `main` entry point function.